docs/config.rst

```diff
@@ -10,11 +10,15 @@ them in order with the latest one taking more priority:
 * ``c:\scrapyd\scrapyd.conf`` (Windows)
 * ``/etc/scrapyd/conf.d/*`` (in alphabetical order, Unix)
 * ``scrapyd.conf``
-* ``~/.scrapyd.conf`` (users home directory)
+* ``~/.scrapyd.conf`` (user's home directory)
 
 The configuration file supports the following options (see default values in
 the :ref:`example <config-example>`).
 
+.. note::
+    The user's home directory is not expanded for the user that may be specified
+    with the ``--uid`` option, but for the user invoking ``scrapyd``.
+    
```

Comment:
TLDR: Why do you need `--uid`? Can we find at which point twistd sets it? 

It looks like you are using scrapyd with `twistd -y`
instead of the provided console script (which uses `--nodaemon` by default).
If we do our own argument parsing in the console script (eg to solve #70)
the console script may start to become the only supported way to run scrapyd.
There's also `scrapyd.script` which I think we should deprecate.

Why do you need to set the uid in the first place?
In case you start as root and drop privileges,
why would you need root?

I can't yet tell at which point twistd drops privileges (provided a uid option)
but if things are as you describe them in the patch, it's definitely after reading the config.

In #186, I want to enable scrapyd to reload the config
without restarting (and killing all running spiders).
I want to first try reading the config after changing the uid
(i.e. render your note false) so that the above feature
will be able to be combined with the uid option.